connector/github/github.go

```diff
@@ -548,7 +552,20 @@ type userEmail struct {
 // The HTTP client is expected to be constructed by the golang.org/x/oauth2 package,
 // which inserts a bearer token as part of the request.
 func (c *githubConnector) userEmail(ctx context.Context, client *http.Client) (string, error) {
+	var (
+		preferredEmailDomainRegexp *regexp.Regexp
+		primaryEmail               userEmail
+		preferredEmails            []userEmail
+	)
+
 	apiURL := c.apiURL + "/user/emails"
+
+	if c.preferredEmailDomain != "" {
+		globExtracted := strings.ReplaceAll(c.preferredEmailDomain, "*", "[a-zA-Z0-9][A-Za-z0-9-][a-zA-Z0-9-]+")
```

Comment:
It seems that the preferred email domain is now a proper regex with a side effect.
Users can provide something like `(.+).my[a-A+].(google|yahoo).com`, and it will be valid.

I'd like to suggest avoiding regex here and using a plain method instead with the following logic (see example here: https://go.dev/play/p/S1X_aGIezCB). With this approach, there are fewer pitfalls, and the method seems more controllable.